src/Microsoft.DotNet.CoreFxTesting/build/core/test/Test.targets

```
@@ -147,26 +147,28 @@
 
   </Choose>
 
-  <ItemGroup>
-    <!-- Add the runner configuration file -->
-    <None Include="$(TestRunnerConfigPath)"
-          CopyToOutputDirectory="PreserveNewest"
-          Visible="false" />
-
-    <!-- Copy test runner to output directory -->
-    <None Include="$([System.IO.Path]::GetDirectoryName('$(XunitConsole472Path)'))\*"
-          Exclude="$([System.IO.Path]::GetDirectoryName('$(XunitConsole472Path)'))\xunit.console.*exe.config;$([System.IO.Path]::GetDirectoryName('$(XunitConsole472Path)'))\xunit.console.x86.exe"
-          Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(XunitConsole472Path)' != ''"
-          CopyToOutputDirectory="PreserveNewest"
-          Visible="false" />
-
-    <_xunitConsoleNetCoreExclude Condition="'$(GenerateDependencyFile)' != 'true'" Include="$([System.IO.Path]::GetDirectoryName('$(XunitConsoleNetCore21AppPath)'))\xunit.console.deps.json" />
-    <None Include="$([System.IO.Path]::GetDirectoryName('$(XunitConsoleNetCore21AppPath)'))\*"
-          Exclude="@(_xunitConsoleNetCoreExclude)"
-          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(XunitConsoleNetCore21AppPath)' != ''"
-          CopyToOutputDirectory="PreserveNewest"
-          Visible="false" />
-  </ItemGroup>
+  <Target Name="CopyRunnerToOutputDirectory" BeforeTargets="CopyFilesToOutputDirectory">
```

Can we instead use DependsOnTargets in `CopyFilesToOutputDirectory`?

---

wouldn't that be too late?

---

Well if we say that `CopyFilesToOutputDirectory` depends on `CopyRunnerToOutputDirectory` it would still run before it.

---

This will go away soon so I'm inclined to leave it as is

---

we have seen a lot of errors in the past in MSBuild where the error isn't propagated when using `BeforeTargets`, with `DependsOn` we guarantee it always propagates errors if so, that is why  I proposed that change. Even if this is temporal there might be failures where we can't copy the runner to output directory and the error is not propagated correctly, hopefully it is not too long before we don't need to copy the runner anymore

---

Just wanted to follow-up on this comment despite the fact that I introduced a regression. The target doesn't invoke any tasks, it only sets items which are copied at a later point in a task which is sequenced inside msbuild/sdk. Therefore I didn't see any necessary action here.

---

Yeah sounds good. Also the target we wanted to run before didn’t have an extensibility point to append to its depends on so the only way was using before.